pkg/crc/network/nameservers.go

```diff
@@ -34,11 +36,35 @@ func GetResolvValuesFromInstance(sshRunner *ssh.Runner) (*ResolvFileValues, erro
 func UpdateResolvFileOnInstance(sshRunner *ssh.Runner, resolvFileValues ResolvFileValues) error {
 	resolvFile, _ := CreateResolvFile(resolvFileValues)
 
-	err := sshRunner.CopyDataPrivileged([]byte(resolvFile), "/etc/resolv.conf", 0644)
-	if err != nil {
-		return fmt.Errorf("Error creating /etc/resolv on instance: %s", err.Error())
+	sd := systemd.NewInstanceSystemdCommander(sshRunner)
+	// Check if ovs-configuration.service exist and if not then it is old bundle and use the same way to
+	// update resolve.conf file
+	if state, err := sd.Status("ovs-configuration.service"); err != nil || state == states.NotFound {
+		err := sshRunner.CopyDataPrivileged([]byte(resolvFile), "/etc/resolv.conf", 0644)
+		if err != nil {
+			return fmt.Errorf("Error creating /etc/resolv on instance: %s", err.Error())
+		}
+		return nil
 	}
-	return nil
+	if err := sd.Start("ovs-configuration.service"); err != nil {
```

Comment:
If I followed correctly, `ovs-configuration.service` will be run during cluster startup as a dependency of `systemctl start kubelet`? We start it manually/explicitly because we want to do things after it is running? for kubelet, this won't matter as it will notice `ovs-configuration` is already running, so nothing more to do ?

Have you explored doing something similar to the suggestion in https://access.redhat.com/solutions/6250271 ?
> The file /etc/resolv.conf can be modified using the nmcli utility via a one-shot systemd service that is (executed after `ovs-configuration.service`)

This would avoid the need for manually starting anything systemd related. Maybe we can unconditionally overwrite resolv.conf, and upload the one shot unit file, and that's it, no need for `sd.Status` or `sd.Start`

Reply:
>  for kubelet, this won't matter as it will notice ovs-configuration is already running, so nothing more to do ?

Yes, that's right.

>Have you explored doing something similar to the suggestion in https://access.redhat.com/solutions/6250271 ?

Yes, this was also suggested in internal channel ,  If we go the machine config way that means the cluster should able to start and api is accessible but if the resolution doesn't work than even our cluster api would respond and that will not work. We can try to have the unit file as template on crc codebase side which we first update and then start but that also means `sd.Status` and `sd.Start` required.

Reply:
I am *not* suggested to use the machine config, I'm only suggesting to use a one-shot systemd service (ie a text file you'd drop in `/etc/systemd/system/` with the right set of `Requires`/`After`/... to have systemd run it automatically after `ovs-configuration.service` but before `kubelet.service`

Reply:
you mean to have this unit file template in the crc codebase and we put that in the instance before kubelet service with updated nameserver and search domain?

Reply:
Yes

Reply:
Can we defer it for next sprint? I want this to be in soon so that we can have it tested for coming release. I can create a follow up issue for same.

Reply:
https://github.com/crc-org/crc/issues/4152